app/scripts/models/reliers/oauth.js

```diff
@@ -213,6 +211,18 @@ var OAuthRelier = Relier.extend({
         }
         throw err;
       });
+
+    function isCorrectRedirect (queryRedirectUri, resultRedirectUri) {
+      if (! queryRedirectUri) {
+        return true;
+      } else if (queryRedirectUri === resultRedirectUri) {
+        return true;
+      } else if (queryRedirectUri === Constants.DEVICE_PAIRING_AUTHORITY_REDIRECT_URI) {
+        return true;
```

Comment:
This seems like a good opportunity to do additional sanity-checks on whether the relier is allowed to use the pairing flow.  If we get here with `queryRedirectUri === Constants.DEVICE_PAIRING_AUTHORITY_REDIRECT_URI` but the client_id is for, say, Pocket, then ISTM that something is going wrong and we should bail out.

One way to achieve this would be to migrate our server-side validation from a single redirect uri to a *list* of allowed redirect URIs for each client, and then we can explicitly put `DEVICE_PAIRING_AUTHORITY_REDIRECT_URI` in the allowlist for the clients that are allowed to use the pairing flow.

But maybe there's something simpler we can do as a first step?

Reply:
`specify a list of client_ids that can use the pairing flow,` sgtm!